perceval/archive.py

```diff
@@ -0,0 +1,312 @@
+        """Load an already existing archive
+
+        :param: fetched_at: string representation of a date identifying an archive
+        """
+
+        archive_path = os.path.join(self.archive_folder_path, fetched_at + STORAGE_EXT)
+
+        if not os.path.exists(archive_path):
+            raise ArchiveError(cause="Archive %s not found!" % fetched_at)
+
+        return Archive(archive_path)
+
+    def new(self):
+        """Create a new archive"""
+
+        archive_path = os.path.join(self.archive_folder_path, now_to_str() + STORAGE_EXT)
+        Archive.create(archive_path)
+        return Archive(archive_path)
+
+    def delete(self, target_date):
+        """Delete an archive identified by its date in folder path
+
+        :param: target_date: string representation of a date identifying an archive
+        """
+
+        path = self.archive_folder_path
+        for f in os.listdir(path):
+            if os.path.isfile(os.path.join(path, f)) and f.startswith(target_date) and f.endswith(STORAGE_EXT):
+                os.remove(os.path.join(path, f))
+                break
+
+    def delete_all(self):
+        """Delete all archives in folder path"""
+
+        path = self.archive_folder_path
+        [os.remove(os.path.join(path, f)) for f in os.listdir(path)
+         if os.path.isfile(os.path.join(path, f)) and f.endswith(STORAGE_EXT)]
+
+    def archives(self, from_date=DEFAULT_DATETIME):
+        """List archives in folder path
+
+        :param from_date: date from when the archives can be retrieved
+
+        :return list of archive names
+        """
+
+        path = self.archive_folder_path
+        archives = [f.replace(STORAGE_EXT, "") for f in os.listdir(path)
+                    if os.path.isfile(os.path.join(path, f)) and f.endswith(STORAGE_EXT)]
+
+        return [os.path.join(path, f + STORAGE_EXT) for f in sorted(archives) if str_to_datetime(f) >= datetime_to_utc(from_date)]
+
+
+class Archive:
+    """Basic storage for Perceval.
+
+    This class allows to archive items for a
+    further recovery. Items are stored in a Sqlite db.
+
+    :param archive_path: path where to load/retrieve the archive
+    """
+
+    def __init__(self, archive_path):
+
+        self.archive_path = archive_path
+
+        self.db = sqlite3.connect(self.archive_path)
+
+    def __del__(self):
+
+        self.db.close()
+
+    @classmethod
+    def create(cls, archive_path):
+        """Create the DB table to store the archive data
+
+        :param archive_path: path where to create an archive
+        """
+        conn = sqlite3.connect(archive_path)
+
+        cursor = conn.cursor()
+        cursor.execute("DROP TABLE IF EXISTS archive")
+        cursor.execute("DROP TABLE IF EXISTS metadata")
+        conn.commit()
+
+        metadata_stmt = "CREATE TABLE metadata ( " \
+                        "origin TEXT, " \
+                        "backend_name TEXT, " \
+                        "backend_version TEXT, " \
```

Comment:
I think we need to include the category of the items fetched during the process.
I'm not sure if we should include `from_date` field this way. There are other possible filters like `offset` that are used in some  backends.

Reply:
why you are not sure? 
I would add the other parameters (offset, etc.) to the table metadata

Reply:
I forgot to mention it. I'm not sure how to store them. I mean, should we store parameters used by `fech` one field each? Or in other hand, one field that stores all parameters in a dict or a JSON structure?

Reply:
I would go for one field each, if there are not many parameters to store